src/components/CurrentUser/CurrentUser.tsx

```diff
@@ -1,12 +1,70 @@
 import React from 'react';
 import './CurrentUser.scss';
+import { User } from '../../react-app-env';
+import { getUser } from '../../api';
 
-export const CurrentUser: React.FC = () => (
-  <div className="CurrentUser">
-    <h2 className="CurrentUser__title"><span>Selected user: 2</span></h2>
+type Props = {
+  userId: number,
+  clearUser: () => void,
+};
 
-    <h3 className="CurrentUser__name">Ervin Howell</h3>
-    <p className="CurrentUser__email">Shanna@melissa.tv</p>
-    <p className="CurrentUser__phone">010-692-6593 x09125</p>
-  </div>
-);
+type State = {
+  user: User | null,
+};
+
+export class CurrentUser extends React.Component<Props, State> {
+  state: State = {
+    user: null,
+  };
+
+  async componentDidMount() {
+    this.loadUserFromServer();
+  }
+
+  async componentDidUpdate(prevProps: Props) {
+    if (prevProps.userId !== this.props.userId) {
+      this.loadUserFromServer();
+    }
+  }
+
+  loadUserFromServer = async () => {
+    try {
+      const user = await getUser(this.props.userId);
+
+      this.setState({ user });
+    } catch (error) {
+      this.setState({ user: null });
+    }
+  };
+
+  render() {
+    const { user } = this.state;
+
+    if (!user) {
+      return (
+        <>
+          <p>Loading User...</p>
+        </>
+      );
+    }
+
+    return (
+      <>
+        {user && (
```

Comment:
Your app won't reach this line of code if previous `!user` statement is true, so this check is redundant
```suggestion
```